cpp/models/abm/world.h

```diff
@@ -186,12 +185,32 @@ class World
     void use_migration_rules(bool param);
     bool use_migration_rules() const;
 
+    /**
+     * adds a testing scheme to the set of all testing schemes for the simulation
+     * @param testing_scheme testing scheme to be added
+     */
+    void add_testing_scheme(const TestingScheme& testing_scheme);
```

Comment:
This implementation allows for multiple testing schemes to (co-)exist. Indeed, one testing scheme is built from multiple testing rules to express different things to be tested. For example, you might want to test all people going to school and also all people above 60 (independent of location) for some reason. This would be covered by two rules within the same testing scheme. Another testing scheme however could test all infected people (independent of the age and location) with a certain probability. This is currently not possible to implement in one scheme without putting the probability in the testing rule. We should discuss this.

Reply:
solved by adding a testingstrategy